scripts/ingest_regress_test_result-new-format.py

```diff
@@ -117,42 +117,46 @@ def ingest_test_result(
-        raw.pop("extra")
-
-        # All allure parameters are prefixed with "__", see test_runner/fixtures/parametrize.py
-        parameters = {
-            p["name"].removeprefix("__"): p["value"]
-            for p in test["parameters"]
-            if p["name"].startswith("__")
-        }
-        arch = parameters.get("arch", "UNKNOWN").strip("'")
-
-        build_type, pg_version, unparametrized_name = parse_test_name(test["name"])
-        labels = {label["name"]: label["value"] for label in test["labels"]}
-        row = Row(
-            parent_suite=labels["parentSuite"],
-            suite=labels["suite"],
-            name=unparametrized_name,
-            status=test["status"],
-            started_at=datetime.fromtimestamp(test["time"]["start"] / 1000, tz=timezone.utc),
-            stopped_at=datetime.fromtimestamp(test["time"]["stop"] / 1000, tz=timezone.utc),
-            duration=test["time"]["duration"],
-            flaky=test["flaky"] or test["retriesStatusChange"],
-            arch=arch,
-            build_type=build_type,
-            pg_version=pg_version,
-            run_id=run_id,
-            run_attempt=run_attempt,
-            reference=reference,
-            revision=revision,
-            raw=json.dumps(raw),
-        )
-        rows.append(dataclasses.astuple(row))
+        try:
+            test = json.loads(f.read_text())
+
+            # Drop unneded fields from raw data
+            raw = test.copy()
+            raw.pop("parameterValues")
+            raw.pop("labels")
+            raw.pop("extra")
+
+            # All allure parameters are prefixed with "__", see test_runner/fixtures/parametrize.py
+            parameters = {
+                p["name"].removeprefix("__"): p["value"]
+                for p in test["parameters"]
+                if p["name"].startswith("__")
+            }
+            arch = parameters.get("arch", "UNKNOWN").strip("'")
+
+            build_type, pg_version, unparametrized_name = parse_test_name(test["name"])
+            labels = {label["name"]: label["value"] for label in test["labels"]}
+            row = Row(
+                parent_suite=labels["parentSuite"],
+                suite=labels["suite"],
+                name=unparametrized_name,
+                status=test["status"],
+                started_at=datetime.fromtimestamp(test["time"]["start"] / 1000, tz=timezone.utc),
+                stopped_at=datetime.fromtimestamp(test["time"]["stop"] / 1000, tz=timezone.utc),
+                duration=test["time"]["duration"],
+                flaky=test["flaky"] or test["retriesStatusChange"],
+                arch=arch,
+                build_type=build_type,
+                pg_version=pg_version,
+                run_id=run_id,
+                run_attempt=run_attempt,
+                reference=reference,
+                revision=revision,
+                raw=json.dumps(raw),
+            )
+            rows.append(dataclasses.astuple(row))
+        except:
```

Comment:
That seems too broad an exception to catch.
I would check that `test` has all the required fields for `Row`, and otherwise skip it

Reply:
Hi @bayandin. No worries there. Please see changes.

Wasn't that straightforward because in order to build `Row` we still need traversing and iterating through various internal properties in the `test` map (arrays and sub-maps).

This should work, I've tested locally with sample test files missing various things. Though its a little complicated for my liking. I prefer the hatchet  🪓 try/catch all errors personally. Less validation = less code to maintain.

Reply:
Alternatively we could limit validation steps to `# 1. Validate JSON` and `# 2. Validate required keys`. Which would be easier to maintain. Or simply `# 2. Validate required keys` in isolation. 

Happy to do that as well. Less code to maintain, though there will still be issues with incorrectly structured JSON throwing errors.